lisa/operating_system.py

```diff
@@ -850,11 +854,65 @@ class Debian(Linux):
     def name_pattern(cls) -> Pattern[str]:
         return re.compile("^debian|Forcepoint|Kali$")
 
-    def add_key(self, server_name: str, key: str) -> None:
-        self._node.execute(
-            f"apt-key adv --keyserver {server_name} --recv-keys {key}",
-            sudo=True,
-        )
+    def add_key(self, server_name: str, key: str = "") -> None:
+        # apt-key add is deprecated starting from Ubuntu 2504.
+        # Use gpg to import the key instead.
+        apt_key_available = False
+        if (
+            self._node.execute("command -v apt-key", shell=True, sudo=True).exit_code
+            == 0
+        ):
+            apt_key_available = True
+
+        if key == "":
+            # if key is not provided, get the key from server_name.
+            wget = self._node.tools[Wget]
+            key_file_path = wget.get(
+                url=server_name,
+                file_path=str(self._node.working_path),
+                force_run=True,
+            )
+            if not apt_key_available:
```

Comment:
use `if apt_key_available/else`. It doesn't need to be a negative condition.

Reply:
Fixed as suggested.